themes/classic/_dev/js/cart.js

```diff
@@ -274,6 +274,18 @@ $(document).ready(() => {
       return false;
     }
   )
+
+  $body.on(
+    'click',
+    '[data-button-action="add-to-cart"]',
+    function () {
+      /**
+       * disable add-to-cart button after first click
+       * this prevents multiple clicks from freezing the modal 
+       */
+      $(this).prop('disabled', true);
```

Comment:
Thanks @rdy4ever this seems legit, I'm just wondering how do you re-enable the button on modal close?

Reply:
Hi @jolelievre 
I'm not doing it, because Prestashop is already doing it. I've tested it and it just works like this. 
Initially, I was planning to use Prestashop's `updatedCart` method to reenable it after cart update (this happens each time the cart is updated, so, technically the bouuton would've been reenabled even before closing the modal), but then I noticed the button is reenabled without this, so I suppose something in core.js does the trick.

Reply:
Ok I got how it happens, when closing the modal the controller `/index.php?controller=product&...` is called and updates the page including the "add to cart" button. So no need to reset it.

However I noticed a similar action is performed in: https://github.com/PrestaShop/PrestaShop/blob/fbd75202b69460805724201068f162e4e371a997/themes/_core/js/cart.js#L81

It's probably better to modify this file, since it's the core prestashop lib it will be available for multiple themes not juste the classic one. So seeing the current code you should disable the button at all time, return from the function when stock is not big enough (the current code). And remember to re-enable the button in case the ajax call fails.

Reply:
I'll dig into this later. You're right, it would be better to solve this in core.